scarlet/source.py

```diff
@@ -178,7 +178,7 @@ def detection_combination(sed, bg_rms, observations, scene, thresh=1):
     """
 
     for c, obs in enumerate(observations):
-        if type(obs).__name__ is not 'ObservationToResample':
+        if type(obs).__name__ is not 'LowResObservation':
```

Comment:
That works, but is not normal way of checking. Please use `if isinstance(obs, LowResObservation) is False`, which will need `from .observation import LowResObservation` at the beginning

Reply:
I also don't think this is what you want to do in the general case of having multiple observations. The logic here is that you want the highest resolution observation to define the morphology, but it may well be the case that none of the observations have the resolution we want to use internally, so they're all LowRes.

Reply:
Finally, the observation for the detection coadd is identified beforehand with `obj_idx`, why is this for loop even necessary?

Reply:
That will be because the whole point of having obs_idx was lost on me. Actually, I can get rid of the whole function and use build_detection_coadd here